### DISCUSSION TITLE
Problem with the communication-generator-rtsp example

### DISCUSSION THREAD

**Discussion by mmcc-xx** [pschatzmann/arduino-audio-tools#812] [Q&A] [answered]
Hi - I'm trying to get the an RTSP server going, and I thought I'd start with the communication-generator-rtsp example. The only thing I changed in the code was the SSID and password.

I'm using a pretty generic ESP32 dev board from Amazon.

As soon as I upload the sketch, the first output in the serial monitor is... 
`Guru Meditation Error: Core  1 panic'ed (Cache disabled but cached memory region accessed). `

Then there is a core dump, the module reboots, and the same thing happens again.

I was able to run streams-generator-serial example successfully.

Any suggestions on how to address this?

Thanks!

**Comment by mmcc-xx** [comment]
After some more investigation, it looks like it is the instantiation of the AudioStreamer that triggers the crashes.

`AudioStreamer streamer = AudioStreamer(&source);`

If I comment that out (and the instantiation of the rtsp object, and the subsequent startup of the rtsp server), the sketch runs successfully. If I then uncomment out just that line, the crashes start happening.

Any suggestions greatly appreciated.

**Comment by mmcc-xx** [comment]
I did some more investigation, and discovered that the RTSP server in communication-generator-rtsp will start up if I create a serial stream.

`CsvStream<int16_t> out(Serial); `

I don't do anything with that serial stream - just instantiate it. No config, no "begin", no copying - just the instantiation. 

That seems super weird.

**Reply by pschatzmann** [reply, marked as answer]
I just made a test and I could not reproduce your issue. However I noticed that this example does not generate a consistent sound any more.

Double check that you have an actual version in the Board Manager. The current version is 2.0.8!
I assume that you installed the latest versions of my libraries...

**Reply by mmcc-xx** [reply]
I’ll double check, but it is a new install of the boards definitions and your libraries. 

I found the examples you have in sandbox for streaming MP3 and AAC via http. The MP3 one failed for me with an error from lame but the AAC example worked, and I think for my application it will work fine in place of RTSP. The input I’m getting from the microphone seems to be very noisy, but that’s just another issue :)

Thanks for looking into this.
